src/app/add_entry/page.jsx

```diff
@@ -0,0 +1,157 @@
+"use client"
+import Link from 'next/link'
+import { supabase } from '../lib/supabase'
+import { useRouter } from 'next/navigation'
+import { UploadButton } from '../utils/uploadthing'
+import { useState } from 'react'
+
+const weatherOptions = [
+  "sunny",
+  "raining",
+  "cloudy"
+]
+
+function AddEntry() {
+  const router = useRouter()
+
+  const [uploadedFileUrls, setUploadedFileUrls] = useState()
+  const [submitDisabled, setSubmitDisabled] = useState(false)
+
+  const handleSubmit = async (e) => {
+    // Prevent page from refreshing on Submit click
+    e.preventDefault()
+
+    const formData = new FormData(e.target);
+    const formProps = Object.fromEntries(formData);
+
+    const submitData = async () => {
+      try {
+        const { error } = await supabase
+          .from('diary_entries')
+          .insert({ date: formProps.date, description: formProps.description, weather: formProps.weather, images: uploadedFileUrls })
+        if (error) {
+          throw error;
+        }
+        alert('Saved!')
+        // Redirect User to List screen after saving
+        router.push('/')
+      } catch (error) {
+        console.error('Error fetching data from Supabase:', error.message);
+      }
+    };
+    submitData();
```

Comment:
unsure why `submitData` is its own function in a function? Doesn't seem necessary 🤔

Reply:
Bonus points would have been given for showing some sort of loading indicator whilst saving 😛